### ISSUE TITLE
[termcolor] Add gray colors

### ISSUE THREAD

**Issue by KodrAus** [BurntSushi/ripgrep#754]
Original issue: https://github.com/sebasmagri/env_logger/issues/59

It would be nice to have a gray variant in the `Color` enum, if it's possible. I think on Windows you can bit-twiddle yourself a gray and dark gray.

cc: @quadrupleslap

**Comment by quadrupleslap** [comment]
It looks like light-gray is a non-intense white, and dark-gray is an intense black.

**Comment by okdana** [comment]
I don't know about Windows, but on UNIX systems, the first 16 ANSI colours are almost always user-configurable, and it's common to have 'white' (`7`) and 'bright black' (`8`) set to shades of grey. This is the way the default palettes of most terminals are set. Sometimes 'black' (`0`) and 'bright white' (`15`) are grey too, though this is less common.

Above the first 16, there are 240 more numbered colours, with `232` to `255` dedicated to various shades of grey. These colours are configurable in some terminals (especially those that take settings from X11), but it's very uncommon for users to mess with them, and in some terminals (like Apple's, AFAIK) they just can't. As a result, it's risky to use them in console applications, because they may clash with the end user's colour theme, or they may be unreadable entirely. Many of the grey colours are illegible on light-background themes, for example.

Then there's also 'true' RGB colour, where you can specify an exact value like in CSS or whatever. These are never configurable, and are thus even more risky for anyone but the end user to mess with.

(On top of that, some terminals don't support 256/RGB colours at all, though all of the major ones do, except the Linux console i think.)

So i don't think it's a good idea to use colours outside of the standard 16 as hard-coded or even default values in something that's intended for other people to use (though there's no harm in giving them the option to do it themselves, obv).

Anyway... `termcolor` would gain support for 256 and RGB ANSI colours if/when PR #452 is merged. It's just been waiting on a more thorough review i think.

**Comment by KodrAus** [comment]
As @quadrupleslap has pointed out, we can get shades of gray using the `set_intense` methods that already exist so I'll go ahead and close this.
